### PR TITLE
write inoperative gin config to tensorboard

### PR DESCRIPTION
**Pull request by witwolf**
#73

### CODE REVIEW

File: alf/utils/common.py [hunk 1]
```diff
@@ -370,16 +370,21 @@ def process(line):
...
     """
-    config_str = gin.operative_config_str()
-    md_config_str = _markdownify_operative_config_str(config_str)
+    operative_config_str = gin.operative_config_str()
+    md_config_str = _markdownify_gin_config_str(operative_config_str)
     tf.summary.text('gin/operative_config', md_config_str)
+    inoperative_config_str = gin_utils.inoperative_config_str()
+    if inoperative_config_str:
+        md_config_str = _markdownify_gin_config_str(inoperative_config_str)
+        tf.summary.text('gin/inoperative_config', md_config_str)
```
Comment (hnyu) [inline alf/utils/common.py]: Maybe write a more descriptive text for `inoperative_config`? For example, "A config string capturing all parameter values configured but not used by the current program (override by explicit call)." So this text is well explained and a person won't get confused even without looking at the code.

Comment (witwolf) [reply alf/utils/common.py]: `in_utils.inoperative_config_str.__doc__` is not appropriate here and explicit str description is too long  

We use gin config with all experiments , and should be familiar with parameters configured at least,   so it should not be a problem ...

Comment (hnyu) [reply alf/utils/common.py]: My point is that people might not be familiar with the concept of "inoperative config" without looking into gin's code. At least we can say several words like "config overridden by program" to make it clear.

Comment (witwolf) [reply alf/utils/common.py]: added at the head  of text summary

File: alf/utils/common.py [hunk 2]
```diff
@@ -379,11 +384,17 @@ def summarize_gin_config():
...
+        operative_config_str,
+        'All parameter values used by configurable functions that are actually called'
+    )
     tf.summary.text('gin/operative_config', md_config_str)
     inoperative_config_str = gin_utils.inoperative_config_str()
     if inoperative_config_str:
-        md_config_str = _markdownify_gin_config_str(inoperative_config_str)
+        md_config_str = _markdownify_gin_config_str(
+            inoperative_config_str,
+            "All parameter values configured but not used by program (override by explicit call)"
```
Comment (emailweixu) [inline alf/utils/common.py]: The configured functions are not called or called by explicit parameter values

**Pull request merged**